aiohttp/streams.py

```diff
@@ -608,8 +606,7 @@ async def read(self) -> _T:
                 raise
 
         if self._buffer:
-            data, size = self._buffer.popleft()
-            self._size -= size
```

Comment:
^

Reply:
Because if we don't delete it we get error in almost all tests the `TypeError error: unsupported operand type(s) for -=: 'int' and 'StreamReader'`. In tests/test_client_functional.py and he refers to this block of code

Reply:
`len(data)`? Obviously you can't subtract a StreamReader from an int...

Though if it's possible for `data` to be something other than bytes, then maybe this won't work.

Reply:
OK, but we will still have tests falling in TestDataQueue. With TypeError error: object of type 'object' has no len(). Now it seems that we need to use `b"something"` instead of `object()` or handle error, but I'm not sure about that.

Reply:
Well, object() is probably unreasonable, so we could maybe update that test... I'll have a look through later.

Reply:
I added types checking to feed_data and now those tests don't crash, but I doubt the correctness of such a solution

Reply:
@Dreamsorcerer could we resume work on this PR?